pkg/apis/migration/v1alpha1/migplan_types.go

```diff
@@ -371,36 +371,51 @@ func (r *MigPlan) BuildRegistryDC(storage *MigStorage, name, dirName string) (*a
 			Namespace: VeleroNamespace,
 		},
 	}
-	err := r.UpdateRegistryDC(storage, deploymentconfig, name, dirName)
+	err := r.UpdateRegistryDC(storage, deploymentconfig, envVars, name, dirName)
 	return deploymentconfig, err
 }
 
-func buildProxyEnvVars() []kapi.EnvVar {
+// Build up proxy env vars for registry DC
+// Returns empty array if secret isn't found or no configuration exists
+func (r *MigPlan) BuildRegistryProxyEnvVars(client k8sclient.Client) []kapi.EnvVar {
 	envVars := []kapi.EnvVar{}
-	if found, s := Settings.HasProxyVar(settings.HttpProxy); found {
+	secret, err := GetSecret(
+		client,
+		&kapi.ObjectReference{
+			Namespace: VeleroNamespace,
+			Name:      ProxySecret,
```

Comment:
What creates the proxy secret?
I'm concerned about the _special_ name.  Usually we use labels.

Reply:
@eriknelson Let's move to attaching a label to this secret. it's fine to keep it `migration-proxy` but lets also add a label.